api/turing/service/ensembler_service.go

```diff
@@ -93,5 +98,5 @@ func (service *ensemblersService) Save(ensembler models.EnsemblerLike) (models.E
 	if err := result.Error; err != nil {
 		return nil, err
 	}
-	return ensembler, nil
+	return service.FindByID(ensembler.GetID(), EnsemblersFindByIDOptions{})
```

Comment:
Why do you call FindByID again? Calling db.Save(ensembler) should mutate the ensembler value, shouldn't it?

Reply:
Also, the error returned by this method will be confusing as a Save method might return a FindByID error.

Reply:
> Why do you call FindByID again? Calling db.Save(ensembler) should mutate the ensembler value, shouldn't it?

Only the fields, that were mutated explicitly (i.e. new `name` is set and then saved), but not for fields, updated automagically (`updated_at`).

Reply:
> Also, the error returned by this method will be confusing as a Save method might return a FindByID error.

That's less of the concern. It was done the same way for other services: https://github.com/gojek/turing/blob/b67c48b7e1b1777f89ec6c6860e3965c516fff9f/api/turing/service/router_service.go#L71

Reply:
Hmm I wonder if gorm allows for the postgres keyword `RETURNING` though. But not consequential

Reply:
No, gorm doesn't support it: https://github.com/go-gorm/gorm/issues/4088